### PR TITLE
Adding cb-clipboard

### PR DESCRIPTION
**Pull request by eunos-1128**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by eunos-1128** [comment]
@Slackadays

Could you agree to being one of the maintainers for this conda package as Clipborad author?

**Comment by Slackadays** [comment]
I can be one of the maintainers

**Comment by eunos-1128** [comment]
> I can be one of the maintainers

Thank you!

**Comment by eunos-1128** [comment]
@conda-forge/help-c-cpp, , ready for review!

**Comment by Slackadays** [comment]
And we thought AI was going to take over the world

File: recipes/cb-clipboard/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+context:
+  name: Clipboard
+  version: "0.9.0.1"
+  author: Slackadays
```
Comment (xhochy) [inline recipes/cb-clipboard/recipe.yaml:4]:
```suggestion
```

File: recipes/cb-clipboard/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    The Clipboard Project (CB for short) is the oh-so-awesome 
+    clipboard manager that's just like a second brain.
+    It's fast and lightweight, feature packed, and super user friendly.
+  repository: https://github.com/${{ author }}/${{ name }}
+  documentation: https://getclipboard.app/docs
+
+extra:
+  recipe-maintainers:
+    - eunos-1128
+    - ${{ author }}
```
Comment (xhochy) [inline recipes/cb-clipboard/recipe.yaml:65]:
```suggestion
    - Slackadays
```

File: recipes/cb-clipboard/build.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+mkdir -p build
+cd build
+
+if [[ "${target_platform}" == "linux-"* ]]; then
+    cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        ..
+elif [[ "${target_platform}" == "osx-"* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=11.0
```
Comment (xhochy) [inline recipes/cb-clipboard/build.sh:17]: Please don't set that manually but rather use the `c_stdlib_version`
```suggestion
```

File: recipes/cb-clipboard/build.bat [hunk 4]
```diff
@@ -0,0 +1,8 @@
+@echo on
+
+mkdir build
+cd build
+
+cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=%PREFIX% ..
+cmake --build . --config Release --parallel %NUMBER_OF_PROCESSORS%
```
Comment (xhochy) [inline recipes/cb-clipboard/build.bat:7]:
```suggestion
cmake --build . --config Release --parallel %CPU_COUNT%
```

File: recipes/cb-clipboard/build.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+export CFLAGS="${CFLAGS/-Os/-O3}"
+export CXXFLAGS="${CXXFLAGS/-Os/-O3}"
```
Comment (xhochy) [inline recipes/cb-clipboard/build.sh:6]: `-Os` should not be part of these flags
```suggestion
```

File: recipes/cb-clipboard/build.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+cd build
+
+if [[ "${target_platform}" == "linux-"* ]]; then
+    cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        ..
+elif [[ "${target_platform}" == "osx-"* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=11.0
+    export SDKROOT=$(xcrun --sdk macosx --show-sdk-path)
```
Comment (xhochy) [inline recipes/cb-clipboard/build.sh:18]: This should be set to a different location by the build tool
```suggestion
```

File: recipes/cb-clipboard/build.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+set -exuo pipefail
+
+export CFLAGS="${CFLAGS/-Os/-O3}"
+export CXXFLAGS="${CXXFLAGS/-Os/-O3}"
+
+mkdir -p build
+cd build
+
+if [[ "${target_platform}" == "linux-"* ]]; then
+    cmake \
```
Comment (xhochy) [inline recipes/cb-clipboard/build.sh:12]:
```suggestion
    cmake $CMAKE_ARGS \
```

File: recipes/cb-clipboard/build.sh [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        ..
+elif [[ "${target_platform}" == "osx-"* ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=11.0
+    export SDKROOT=$(xcrun --sdk macosx --show-sdk-path)
+    
+    cmake \
```
Comment (xhochy) [inline recipes/cb-clipboard/build.sh:20]:
```suggestion
    cmake $CMAKE_ARGS \
```

File: recipes/cb-clipboard/recipe.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+      then:
+        - alsa-lib
+        - xorg-libx11 
+        - wayland 
+  run:
+    - if: linux
+      then:
+        - alsa-lib
+        - xorg-libx11
+        - wayland
```
Comment (xhochy) [inline recipes/cb-clipboard/recipe.yaml:44]: They should come in via their `run_exports`
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/cb-clipboard/build.bat [hunk 10]
```diff
@@ -0,0 +1,10 @@
+@echo on
+
+mkdir build
+cd build
+
+cmake -DCMAKE_BUILD_TYPE=Release ^
+    -DOPENSSL_ROOT_DIR=%PREFIX%\Library ^
+    -DCMAKE_INSTALL_PREFIX=%PREFIX% ..
```
Comment (xhochy) [inline recipes/cb-clipboard/build.bat:8]:
```suggestion
    -DOPENSSL_ROOT_DIR=%LIBRARY_PREFIX%^
    -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ..
```

File: recipes/cb-clipboard/build.sh [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        ..
+elif [[ "${target_platform}" == "osx-"* ]]; then
+
+    cmake ${CMAKE_ARGS} \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        -DCMAKE_OSX_ARCHITECTURES=${OSX_ARCH} \
+        -DCMAKE_OSX_DEPLOYMENT_TARGET=10.15 \
```
Comment (xhochy) [inline recipes/cb-clipboard/build.sh:21]: Don't set that explicitly, use `c_stdlib_version`
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by Slackadays** [comment]
@eunos-1128 you can do it! 🥇

**Comment by eunos-1128** [comment]
I replaced `recipe.yaml` with `meta.yaml` to use conda-build instead of rattler-build because MacOS build always fails even with the config file defining `c_stdlib_version`.

**Comment by Slackadays** [comment]
@xhochy The fixes have been implemented

File: recipes/cb-clipboard/build.sh [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    cmake ${CMAKE_ARGS} \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        ..
+elif [[ "${target_platform}" == "osx-"* ]]; then
+    cmake ${CMAKE_ARGS} \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+        -DCMAKE_OSX_ARCHITECTURES=${OSX_ARCH} \
+        -DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET} \
```
Comment (xhochy) [inline recipes/cb-clipboard/build.sh:20]: Please don't set this explicitly, this should come via `CMAKE_ARGS`
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**